plugins/woocommerce/src/Database/Migrations/CustomOrderTable/CLIRunner.php

```diff
@@ -0,0 +1,412 @@
+	 *
+	 * @var PostsToOrdersMigrationController
+	 */
+	private $post_to_cot_migrator;
+
+	/**
+	 * Init method, invoked by DI container.
+	 *
+	 * @param CustomOrdersTableController      $controller Instance.
+	 * @param DataSynchronizer                 $synchronizer Instance.
+	 * @param PostsToOrdersMigrationController $posts_to_orders_migration_controller Instance.
+	 *
+	 * @internal
+	 */
+	final public function init( CustomOrdersTableController $controller, DataSynchronizer $synchronizer, PostsToOrdersMigrationController $posts_to_orders_migration_controller ) {
+		$this->controller           = $controller;
+		$this->synchronizer         = $synchronizer;
+		$this->post_to_cot_migrator = $posts_to_orders_migration_controller;
+	}
+
+	/**
+	 * Registers commands for CLI.
+	 */
+	public function register_commands() {
+		WP_CLI::add_command( 'wc cot count', array( $this, 'count' ) );
+		WP_CLI::add_command( 'wc cot migrate', array( $this, 'migrate' ) );
+		WP_CLI::add_command( 'wc cot verify_cot_data', array( $this, 'verify_cot_data' ) );
+	}
+
+	/**
+	 * Check if the COT feature is enabled.
+	 *
+	 * @param bool $log Optionally log a error message.
+	 *
+	 * @return bool Whether the COT feature is enabled.
+	 */
+	private function is_enabled( $log = true ) {
+		if ( ! $this->controller->is_feature_visible() ) {
+			if ( $log ) {
+				WP_CLI::log(
+					sprintf(
+						// translators: %s - link to testing instructions webpage.
+						__( 'Custom order table usage is not enabled. If you are testing, you can enable it by following the testing instructions in %s', 'woocommerce' ),
+						'https://developer.woocommerce.com/' // TODO: Change the link when testing instructin page is live.
+					)
+				);
+			}
+		}
+
+		return $this->controller->is_feature_visible();
+	}
+
+	/**
+	 * Helper method to log warning that feature is not yet production ready.
+	 */
+	private function log_production_warning() {
+		WP_CLI::log( __( 'This feature is not production ready yet. Make sure you are not running these commands in your production environment.', 'woocommerce' ) );
+	}
+
+	/**
+	 * Count how many orders have yet to be migrated into the custom orders table.
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     wp wc cot count
+	 *
+	 * @param array $args Positional arguments passed to the command.
+	 *
+	 * @param array $assoc_args Associative arguments (options) passed to the command.
+	 *
+	 * @return int The number of orders to be migrated.*
+	 */
+	public function count( $args = array(), $assoc_args = array() ) {
+		if ( ! $this->is_enabled() ) {
+			return 0;
+		}
+
+		// phpcs:ignore WordPress.DB.PreparedSQL.NotPrepared
+		$order_count = $this->synchronizer->get_current_orders_pending_sync_count();
+
+		$assoc_args = wp_parse_args(
+			$assoc_args,
+			array(
+				'log' => true,
+			)
+		);
+		if ( isset( $assoc_args['log'] ) && $assoc_args['log'] ) {
+			WP_CLI::log(
+				sprintf(
+				/* Translators: %1$d is the number of orders to be migrated. */
```

Comment:
Minor formatting thing:

```suggestion
					/* Translators: %1$d is the number of orders to be migrated. */
```

Funnily enough, this also almost always happens to me when adding translator notes. Not sure why, perhaps a PhpStorm glitch.